ttnn/cpp/ttnn/operation.hpp

```diff
@@ -556,8 +577,10 @@ struct DeviceOperation final {
                 const auto& operation = *reinterpret_cast<const std::decay_t<T>*>(&storage);
                 if constexpr (detail::implements_create_output_tensors_with_optional_output_tensors<T>()) {
                     return operation.create_output_tensors(input_tensors, output_tensors);
-                } else {
+                } else if constexpr (detail::implements_create_output_tensors<T>()) {
                     return operation.create_output_tensors(input_tensors);
+                } else {
+                    return detail::default_create_output_tensors(operation, input_tensors);
```

Comment:
allows to remove plenty of code potentially?

Reply:
yes

Reply:
If op implements any variation of `create_output_tensors`, we can make it entirely custom and not have infra do anything. I think today, `generic_create_output_tensors` does too little to justify it as infra. I think it could actually just be a helper. And ops don't even need to implement `compute_output_shapes`